tools/clang/tools/dxil2spv/lib/dxil2spv.cpp

```diff
@@ -361,6 +376,34 @@ void Translator::createThreadIdInstruction(llvm::CallInst &instruction) {
   instructionMap[&instruction] = loadInstr;
 }
 
+void Translator::createBinaryOpInstruction(llvm::BinaryOperator &instruction) {
+  llvm::Instruction::BinaryOps opcode = instruction.getOpcode();
+  spirv::SpirvBinaryOp *result;
+
+  switch (opcode) {
+  // Shift left instruction.
+  case llvm::Instruction::Shl: {
+    // Value to be shifted.
+    spirv::SpirvInstruction *val = instructionMap[instruction.getOperand(0)];
```

Comment:
Do you think we have to add an error check here?
I am just worried the case we don't have a key `instruction.getOperand(0)` in `instructionMap`.